src/main/java/com/amazon/opendistroforelasticsearch/security/user/User.java

```diff
@@ -263,4 +263,8 @@ public final void addOpenDistroSecurityRoles(final Collection<String> securityRo
     public final Set<String> getOpenDistroSecurityRoles() {
         return this.openDistroSecurityRoles == null ? Collections.emptySet() : Collections.unmodifiableSet(this.openDistroSecurityRoles);
     }
+
+    public final String getUserRolesString() {
+        return name + "|" + String.join(",", getRoles()) + "|" + String.join(",", getOpenDistroSecurityRoles());
```

Comment:
Can roles or odfe roles contain `,`?

Reply:
As far as I know, it can't contain `,`. Pipe separated string format is used at other places including : UserInjector, RolesInjector.